internal/containerupdate/exec_updater_test.go

```diff
@@ -0,0 +1,124 @@
+package containerupdate
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/windmilleng/tilt/internal/k8s"
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/testutils"
+)
+
+var toDelete = []string{"/foo/delete_me", "/bar/me_too"}
+var (
+	cmdA = model.Cmd{Argv: []string{"a"}}
+	cmdB = model.Cmd{Argv: []string{"b", "bar", "baz"}}
+)
+var cmds = []model.Cmd{cmdA, cmdB}
+
+func TestUpdateContainerDoesntSupportRestart(t *testing.T) {
+	f := newExecFixture(t)
+
+	err := f.ecu.UpdateContainer(f.ctx, TestDeployInfo, newReader("boop"), toDelete, cmds, false)
+	if assert.NotNil(t, err, "expect Exec UpdateContainer to fail if !hotReload") {
+		assert.Contains(t, err.Error(), "ExecUpdater does not support `restart_container()` step")
+	}
+}
+
+func TestUpdateContainerDeletesFiles(t *testing.T) {
+	f := newExecFixture(t)
+
+	// No files to delete
+	err := f.ecu.UpdateContainer(f.ctx, TestDeployInfo, newReader("boop"), nil, cmds, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, call := range f.kCli.ExecCalls {
+		if len(call.Cmd) >= 1 && call.Cmd[0] == "rm" {
+			t.Errorf("found kubernetes exec `rm` call, expected none b/c no files to delete")
+			t.Fail()
+		}
+	}
+
+	// Two files to delete
+	err = f.ecu.UpdateContainer(f.ctx, TestDeployInfo, newReader("boop"), toDelete, cmds, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var rmCmd []string
+	for _, call := range f.kCli.ExecCalls {
+		if len(call.Cmd) >= 1 && call.Cmd[0] == "rm" {
+			rmCmd = call.Cmd
+			break
```

Comment:
instead of breaking I might put above this an assertion that rmCmd is nil, to catch the case where we erroneously have two rm cmds

either that or change it to rmCmds, keep a list of all of them, and then assert.Equals on the list (which could later lend itself better to like a `func (kCli *Client) GetExecCalls(cmd string) [][]string` and `rmCmds := f.kCli.GetExecCalls("rm")`)